src/ui/replay_parser.rs

```diff
@@ -2366,6 +2367,192 @@ impl ToolkitTabViewer<'_> {
         }
     }
 
+    fn generate_game_data(report: &BattleReport, ui_report: &UiReport, replay_file: &ReplayFile) -> serde_json::Value {
+        let players = ui_report.vehicle_reports.iter().map(|report| {
+            let damage_dealt_total = report.actual_damage.unwrap_or(0);
+            let potential_damage_total = report.potential_damage.unwrap_or(0);
+            let damage_received_total = report.received_damage.unwrap_or(0);
+
+            // Build damage dealt section
+            let mut damage_dealt = serde_json::Map::new();
+            
+            // Always include total
+            damage_dealt.insert("damage_total".to_string(), json!(damage_dealt_total));
+            
+            // Only include non-None values (don't include nulls at all)
+            if let Some(damage_report) = &report.actual_damage_report {
+                if let Some(v) = damage_report.ap { damage_dealt.insert("damage_ap".to_string(), json!(v)); }
+                if let Some(v) = damage_report.he { damage_dealt.insert("damage_he".to_string(), json!(v)); }
+                if let Some(v) = damage_report.he_secondaries { damage_dealt.insert("damage_he_secondary".to_string(), json!(v)); }
+                if let Some(v) = damage_report.sap { damage_dealt.insert("damage_sap".to_string(), json!(v)); }
```

Comment:
For your use case is there a reason why you only want fields if they have a value present? I prefer to keep the schema consistent but just have `null` values in the JSON if the values don't exist.